corehq/apps/couch_sql_migration/management/commands/migrate_domain_from_couch_to_sql.py

```diff
@@ -77,13 +90,15 @@ def handle(self, domain, **options):
                 )
             set_couch_sql_migration_not_started(domain)
             _blow_away_migration(domain)
+
         if options['stats_short'] or options['stats_long']:
             self.print_stats(domain, short=options['stats_short'])
         if options['show_diffs']:
             self.show_diffs(domain)
+
         if options['COMMIT']:
             self.require_only_option('COMMIT', options)
-            assert couch_sql_migration_in_progress(domain)
+            assert couch_sql_migration_in_progress(domain, any_run=False)
```

Comment:
Assertion errors should not be used to indicate bad user input. They are intended to indicate that there is an inconsistency in logic of the program itself, and should not be able to be triggered by bad user input (unless the bad user input exposes a bug in the internal logic of the program).

One way to indicate errors like this is to use something like [`ArgumentParser.error`](https://docs.python.org/2/library/argparse.html#argparse.ArgumentParser.error)

I realize I didn't catch this earlier when this and `require_only_option` was originally written. Sorry about that.

Also, it would be nice to have an error message to clearly explain what went wrong. Something like "Cannot commit dry run migration" (although I'm not sure that's what a failure here means)

Reply:
55f6e15

For a management command, especially one only I'm going to run, I'm really not too worried about this/being pythonic. I'm expecting to look at the stacktrace anyway.

`CommandError` seems to be django's built in version of `ArgumentParser.error`, and I will say it's nice having that printed red error though.